src/components/LineChart/LineChart.jsx

```diff
@@ -96,18 +195,100 @@ export default class LineChart extends PureComponent {
     this.lines = this.g.append('g').classed('lines-group', true);
     this.circles = this.g.append('g').classed('circles-group', true);
 
+    // container for showing the x highlighte date indicator
+    this.highlightDate = this.g.append('g').attr('class', 'highlight-date');
+    this.highlightDate.append('line')
+      .attr('x1', 0)
+      .attr('x2', 0)
+      .attr('y1', 0)
+      .attr('y2', innerHeight + 3)
+      .attr('class', 'highlight-ref-line');
+
+    // container for showing the highlighted line
+    this.highlightLine = this.g.append('g').attr('class', 'highlight-line');
+    this.highlightLine.append('rect').attr('class', 'series-overlay');
+    this.highlightLine.append('g');
+
+    // add in mouse listener -- this should be added above the lines and axes
+    const that = this;
+    this.mouseListener = this.g.append('rect')
+      .attr('class', 'mouse-listener')
+      .style('fill', '#f00')
+      .style('stroke', 'none')
+      .style('opacity', 0)
+      .on('mousemove', function mouseMoveListener() {
+        that.onMouseMove.call(that, d3.mouse(this));
+      })
+      .on('mouseout', () => this.onMouseMove(null));
+
+
     this.update();
   }
 
   /**
+   * Helper function to make components necessary for the legend
+   */
+  makeLegendComponents(series = [], annotationSeries = [], legendWidth) {
```

Comment:
Yea that's partially why I separated this out like this. There's some complexity around it though since we need to know the height of the legend to make room for it in the chart and that's tough when it is separate. One way is to compute the height in this chart via some shared function that the legend also uses, but it doesn't seem like the best plan. Since we have to have it all in one SVG for PNG export we can't just use the DOM to flow :(